odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -105,15 +111,16 @@ def _check_sql_injection_risky(self, node):
             current = node
             while (current and not isinstance(current.parent, astroid.FunctionDef)):
                 current = current.parent
-            parent = current.parent
-
-            # 2) check how was the variable built
-            for node_ofc in parent.nodes_of_class(astroid.Assign):
-                if node_ofc.targets[0].as_string() != first_arg.as_string():
-                    continue
-                is_concatenation = self._check_concatenation(node_ofc.value)
-                if is_concatenation:
-                    break
+            if current:
```

Comment:
I took advantage to fix an issue related with the following code:

```python
queries = [
    "SELECT id FROM res_partner",
    "SELECT id FROM res_users",
]
for query in queries:
    self.env.cr.execute(query)
```



The check sql-injection shows the following error:
  - AttributeError: 'NoneType' object has no attribute 'parent'

So, Now it is validating if it is not None

Reply:
cc @xmo-odoo